src/USER-ESFRICTION/fix_esfriction.cpp

```diff
@@ -0,0 +1,203 @@
+  // Compute total energy lost due to ES as per NRT model
+  double E_hat = get_ehat();
+  ES_loss_NRT = E_pka - E_hat;
+  // following computes coefficient of friction (beta) using LS model
+  double number_density = atom->natoms / 
+    (domain->xprd * domain->yprd * domain->zprd) * 1e30; // number-per-m3
+  double E_0 = h_cross*h_cross / (2*m_e) * 
+    pow( 3*pi*pi*Ne_val*(number_density), 2.0/3.0) * joules_to_ev; // eV
+  double v_0 = 5.931e5 * sqrt(E_0) * 1e-2; // ang/ps
+  // lambda is in eV-Angstrom-ps units
+  double lambda = 8.0 * pi * (e_charge * e_charge * 10) * bohr_radius * 
+    pow(Z1, 1.0/6.0) * Z1*Z2 / ( v_0 * pow( pow(Z1, 2.0/3.0) + 
+    pow(Z2, 2.0/3.0) , 3.0/2.0) ); 
+  beta = number_density * lambda / 1.0e30; //eV-ps-angstrom^-2 units
+  // Init energy lost due to ES as per LS model (calculated numerically)
+  ES_loss_LS = 0.0;
+  MPI_Comm_rank(world, &my_rank);
+  if (my_rank==0){
+    if(screen){
+      fprintf(screen, "E_pka = %g eV\n", E_pka);
+      fprintf(screen, "Energy loss due to ES (from NRT model) = %g\n", 
+        ES_loss_NRT);
+    }
+    if (logfile){
+      fprintf(logfile, "E_pka = %g eV\n", E_pka);
+      fprintf(logfile, "Energy loss due to ES (from NRT model) = %g\n", 
+        ES_loss_NRT);
+    }
+  }
+}
+
+/* ---------------------------------------------------------------------- */
+
+double FixESFriction::get_kenergy(double A, double vx, double vy, double vz)
+{
+  double gms_to_kg = 1e-3;  // grams to kg
+  double apps_to_mps = 100; // amstrong per picoseconds -to- m/s
+  return (0.5 * A * ( vx*vx + vy*vy + vz*vz ) * gms_to_kg * apps_to_mps * 
+    apps_to_mps * joules_to_ev / Nav); // in eV
+}
+
+/* ----------------------------------------------------------------------
+   Ehat is total energy lost due to ES based on NRT model 
+----------------------------------------------------------------------- */
+
+double FixESFriction::get_ehat()
+{
+  double Emax = 25.0 * pow(Z1, 4.0/3.0) * A1;
+  if (E_pka >= Emax){
+    if (screen){
+      fprintf(screen, "E_pka >= Emax (%g > %g) in esfriction fix.\n", E_pka, 
+        Emax);
+      fprintf(screen, "Hence E_hat calculation (NRT) might be invalid.\n");
+    }
+    if (logfile){
+      fprintf(logfile, "E_pka >= Emax (%g > %g) in esfriction fix.\n", E_pka, 
+        Emax);
+      fprintf(logfile, "Hence E_hat calculation (NRT) might be invalid.\n");
+    }
+  }
+  double eps, gofeps, kay;
+  double const1 = pow( (9.0 * pi*pi / 128.0), (1.0/3.0) );
+  double a = ( const1 * (bohr_radius /  10.0) ) / sqrt( pow(Z1, 2/3.0) + 
+    pow(Z2, 2/3.0) ) ;
+  eps = ( A2 / (A1+A2) ) * ( (a / ( Z1 * Z2 * e_charge * e_charge ) ) ) * 
+    E_pka;
+  gofeps = (3.4008 * pow(eps, 1/6.0)) + (0.40244 * pow(eps, 3.0/4.0)) + eps;
+  kay = 0.1337 * pow(Z1, 1/6.0) * sqrt(Z1/A1);
+  double Ehat = E_pka / (1.0 + kay * gofeps);
+  return Ehat;
+}
+
+/* ---------------------------------------------------------------------- */
+FixESFriction::~FixESFriction()
```

Comment:
An empty destructor can be defined in the class definition directly. No need to have it here.